R/01-createDuplicateModal.R

```diff
@@ -0,0 +1,131 @@
+#' Creates a modalDialog for detectDuplicates module
+#'
+#' @param vars dataframe column names
+#' @param session current session
+createDuplicateModal <- function(vars, session) {
+  ns <- session$ns
+  modalDialog(
+    title = "Duplicate Identifier",
+    easyClose = TRUE,
+    size = "l",
+    footer = modalButton("Close (Without Saving Changes)"),
+    tagList(
+      HTML("This section gives you the option to find and export duplicates or remove them from the dataset.
+                You must first select the columns in which you want to search for duplicates.
+                You can then select the degree of similarity, independently for each column, which specifies how similar cases must be to be classified as duplicates.<br><br>
+                For numeric columns, you can choose between 'Exact Match' and 'Rounded Match'.
+                For the 'Exact Match' selection, observations are only classified as duplicated if the values match exactly.
+                For 'Rounded Match', values are classified as identical if they are identical after rounding.
+                For this reason, you can specify a rounding factor if you have selected 'Rounded Match'.
+                A value of 2, for example, means that the values are rounded to 2 decimal places before being checked for equality.
+                A value of 0 means that the values are rounded to whole numbers.
+                You can also enter negative numbers, which will result in rounding to the power of ten, e.g. the value -2 rounds to the nearest hundred.<br>
+                For text columns you can choose between 'Case Sensitive', where a distinction is made between upper and lower case letters and
+                'Case Insensitive', where no distinction is made.<br><br>
+                You then have the option to display the duplicates and/or remove them from the dataset."),
+      hr(),
+      selectizeInput(
+        inputId = ns("variables"),
+        label = "Duplicate Identifikation Columns",
+        choices = vars,
+        multiple = TRUE
+      ),
+      hr(),
+      fluidRow(
+        column(
+          4,
+          selectizeInput(
+            inputId = ns("selectedVariable"),
+            label = "Degree of Similarity for Variable...",
```

Comment:
I think it could help here to have more options in the select input, such as in 
![image](https://user-images.githubusercontent.com/16759098/215454050-039bfc0d-6c54-44c6-8d24-b140dcaabe97.png)

You can compare here: https://github.com/Pandora-IsoMemo/iso-app/blob/61733904472fb6d3449c452b761a0830f7a05a8f/R/03-dataExplorer.R#L156-L159

Reply:
Great idea, thank you.